addons/purchase/models/purchase.py

```diff
@@ -706,25 +713,45 @@ def _confirm_reception_mail(self):
                 order.message_post(body="The order receipt has been acknowledged by %s." % order.partner_id.name)
 
     def _update_date_planned_for_lines(self, updated_dates):
-        if updated_dates:
-            note = self._compose_note(updated_dates)
-            self.activity_schedule(
+        # creat or update the activity
+        self._create_update_date_activity(updated_dates)
+        # update the date on PO line
+        for line, date in updated_dates:
+            line._update_date_planned(date)
+
+    def _create_update_date_activity(self, updated_dates):
+        """Helper method for creating or updating an activity for the user when
+        vendor update scheduled date on portal website."""
+        activity = self.env['mail.activity'].search([
+            ('summary', '=', 'Date Updated'),
+            ('res_model_id', '=', self.env['ir.model']._get(self._name).id),
+            ('res_id', '=', self.id),
+            ('user_id', '=', self.user_id.id)], limit=1)
+        if activity:
+            for line, date in updated_dates:
+                # replace special characters in product_name by '.'
+                product_name_re = re.sub('[\[\]\(\)\@\$\?\{\}\|\*\.\+\/\\\]', '.', line.product_id.display_name)
```

Comment:
@nim-odoo I think it is complexe, no ?

Reply:
@fmdl I'm not in charge of this reviewing this task, this might still even be in development.

Reply:
@sle-odoo Any good idea about this?